android/src/main/java/com/dooboolab/RNIap/RNIapAmazonListener.java

```diff
@@ -199,6 +199,14 @@ public void onPurchaseUpdatesResponse(
         if (response.hasMore()) {
           PurchasingService.getPurchaseUpdates(false);
         } else {
+          if (purchases.size() > 0 && promiseItem != null) {
+            DoobooUtils
+              .getInstance()
+              .resolvePromisesForKey(
+                RNIapAmazonModule.PROMISE_BUY_ITEM,
+                promiseItem
+              );
+          }
```

Comment:
Don't you want to `return` here to prevent continuing code execution?

Reply:
No, in this case, we want to potentially resolve all three possible promises that triggered this `onPurchaseUpdatesResponse` response from the Amazon SDK. (there are 2 resolves right after this)

Reply:
Do you mean you want to resolve them at once? Only one event should resolve one promise at the same time.

Reply:
Maybe I am just naive, but why not? 

If 3 separate promises have been created, and are pending, and this single callback from the Amazon SDK contains the necessary data to resolve them, why not resolve all of them at that point?

Also, there are already 2 resolves in the code immediately after this, so if this really is an issue, those should be changed as well.

Reply:
@curiousdustin I think you've thought like below.

![image](https://user-images.githubusercontent.com/27461460/116848826-70702880-ac28-11eb-8d0a-6562b24dc740.png)

While I think is below.
![image](https://user-images.githubusercontent.com/27461460/116848836-76660980-ac28-11eb-9b6e-a5e84d8497d1.png)

When you call the `buyItemByType`, it opens the promise to be resolved. However, when other thread tries to resolve promise if not opened, I am afraid some miss leading behavior will occur.

Secondly, opening a promise to resolve then if 3 promises are ready to be resolved at a single operation, it doesn't look right to me.

Reply:
@curiousdustin I was going to update these changes myself. However, I am also not sure how this is going to cause the side-effect. Currently, I've changed my mind to accept this `PR`.

Let's see who's coming after this 🙏